ci: add import-time gate for rogue (<= 1.0s best-of-5 cold imports)

MANASAB805 opened the pull request:
## Summary

Add a CI job that measures the cold import time of the `rogue` package and fails the build if the best-of-5 import time exceeds 1.0s.

This is implemented as a new workflow: [.github/workflows/import-time.yml](cci:7://file:///Users/manasabalakrishhna/CascadeProjects/rogue/.github/workflows/import-time.yml:0:0-0:0).

## Motivation

We’ve invested in import-time optimizations (e.g., `if TYPE_CHECKING`, deferring heavy imports) and want to prevent regressions. Consistently enforcing an upper bound on import time helps preserve fast startup, especially for CLI/TUI entry points and server processes.

## What’s Changed

- New GitHub Actions workflow [import-time.yml](cci:7://file:///Users/manasabalakrishhna/CascadeProjects/rogue/.github/workflows/import-time.yml:0:0-0:0):
  - Triggers on `push` and `pull_request`.
  - Installs the package from the PR via `pip install -e .`.
  - Measures cold import time by spawning a fresh Python subprocess 5 times.
  - Uses the best (minimum) time to reduce runner noise.
  - Fails if the best time exceeds `THRESHOLD_SECONDS` (default 1.0s).

## Implementation Details

- “Cold” import is approximated by running:
  - `python -c "import importlib, time; t=time.perf_counter(); importlib.import_module('rogue'); print(time.perf_counter()-t)"` in a fresh subprocess 5 times.
- The job prints the best measured import time and exits with non-zero status if above threshold.
- Threshold is configured via `THRESHOLD_SECONDS` env var (defaults to `1.0`).

## Trade-offs

- This implementation uses `pip install -e .` in CI for simplicity and isolation of the PR’s code. If project-wide CI is standardized on `uv`, we can switch to `uv run` equivalents in a follow-up without changing the measurement approach.
- CI runner variability can introduce some noise. Using the best-of-5 minimizes flakes while still catching true regressions.

## How to Test Locally

```bash
# Cold import timing (example local check)
python -c "import importlib, time; t=time.perf_counter(); importlib.import_module('rogue'); print(time.perf_counter()-t)"

MANASAB805 commented:
Hi @drorIvry @yuval-qf — please approve CI workflow runs for this PR.

This adds [.github/workflows/import-time.yml](cci:7://file:///Users/manasabalakrishhna/CascadeProjects/rogue/.github/workflows/import-time.yml:0:0-0:0) to enforce a 1.0s max best-of-5 cold import time for `rogue`. It installs the package from the PR and measures via subprocess imports to reduce cache effects.

Happy to switch the install to `uv` to align with the repo’s tooling if preferred.
Fixes qualifire-dev/rogue#69

yuval-qf reviewed (changes requested):
Thanks for this PR!
Left a few comments

MANASAB805 commented:
Hi @yuval-qf, thanks for the review — I’ve addressed your comments.

- Install flow aligned with repo using uv + venv and local SDK:
  - uv venv
  - uv pip install -e .
  - uv pip uninstall -y rogue-ai-sdk || true
  - uv pip install -e sdks/python --force-reinstall
- Timing approach updated: /usr/bin/time + bc, best-of-5 (Bash).
- Python version sourced from .python-version via actions/setup-python.

Changes in /.github/workflows/import-time.yml (latest commit: eed5cbf).

If you’d like this to trigger only on PRs (not pushes), I can scope on: to pull_request. I can also sync the branch with main if preferred.
Please re-review when you have a moment.
